cpp/src/community/detail/mis_impl.cuh

```diff
@@ -102,10 +92,7 @@ rmm::device_uvector<vertex_t> compute_mis(
   out_degrees.resize(0, handle.get_stream());
```

Comment:
```
  thrust::for_each(
    handle.get_thrust_policy(),
    vertex_begin,
    vertex_end,
    [out_degrees = raft::device_span<edge_t const>(out_degrees.data(), out_degrees.size()),
     ranks       = raft::device_span<vertex_t>(ranks.data(), ranks.size()),
     v_first     = graph_view.local_vertex_partition_range_first()] __device__(auto v) {
      auto v_offset = v - v_first;
      if (out_degrees[v_offset] == 0) { ranks[v_offset] = std::numeric_limits<vertex_t>::lowest(); }
    });
```

You can use `thrust::transform_if` for this. It will be more specific than for_each (one of the RAPIDS reviewer mentioned `for_each` as glorified for loop, better avoid it if there exists a better matching thrust algorithm).

Reply:
`transform_if` much better documents the intention (transform to std::numeric_limits<vertex_t>::lowest() if the degree is 0) without comments than `for_each`.